server/server.go

```diff
@@ -32,49 +34,51 @@ func StartHttpServer(opts *options.Options) error {
 	return nil
 }
 
-// Check that we can open a TPC connection to all the ports in opts.Ports
-func checkTcpPorts(opts *options.Options) *httpResponse {
+func checkHealthChecks(opts *options.Options) *httpResponse {
 	logger := opts.Logger
 	logger.Infof("Received inbound request. Beginning health checks...")
 
-	allPortsValid := true
-
+	// initialize failedChecks to 0
+	var failedChecks uint64
 	var waitGroup = sync.WaitGroup{}
 
-	for _, port := range opts.Ports {
+	for _, tcpCheck := range opts.Checks.TcpChecks {
+		name := tcpCheck.Name
+		host := tcpCheck.Host
+		port := tcpCheck.Port
 		waitGroup.Add(1)
 		go func(port int) {
-			err := attemptTcpConnection(port, opts)
+			err := attemptTcpConnection(tcpCheck.Host, tcpCheck.Port,opts)
 			if err != nil {
-				logger.Warnf("TCP connection to port %d FAILED: %s", port, err)
-				allPortsValid = false
+				logger.Warnf("TCP connection to %s at %s:%d FAILED: %s", name, host, port, err)
+				atomic.AddUint64(&failedChecks, 1)
 			} else {
-				logger.Infof("TCP connection to port %d successful", port)
+				logger.Infof("TCP connection to %s at %s:%d successful", name, host, port)
 			}
-
 			waitGroup.Done()
```

Comment:
I know it's not your code, but typically, `waitGroup.Done()` should be in a `defer` to ensure it's always called